signetdev/host/signetdev_emulate.c

```diff
@@ -558,7 +566,7 @@ int signetdev_emulate_init(const char *filename)
 		}
 		fseek(g_deviceState.db_file, SEEK_SET, 0);
 		size_t rc = fread(g_deviceState.device_data, signetdev_device_block_size(), signetdev_device_num_storage_blocks(), g_deviceState.db_file);
-		if (rc != signetdev_device_num_storage_blocks()) {
+		if (rc != (unsigned int)signetdev_device_num_storage_blocks()) {
```

Comment:
Looks like another case where probably the function should have been "unsigned int" to begin with

Reply:
Yes, perhaps. But in the client code often signed int is expected from these functions. Then we need to cast there. It's like whack-a-mole.